### PR TITLE
Feature/add diagrams page

### LINKED ISSUE #239: Integrate plugin-core version 0.16.0
Once [Plugin-core version 0.16.0](https://github.com/ditrit/leto-modelizer-plugin-core/milestone/7) is released, we can integrated it in `leto-modelizer`.

### LINKED ISSUE #240: Integrate terrator-plugin version 0.3.0
Once [Terrator-plugin version 0.3.0](https://github.com/ditrit/terrator-plugin/milestone/1) is released, we can integrated it in `leto-modelizer`.

### LINKED ISSUE #241: Integrate Githubator-plugin version 0.2.0
Once [Githubator-plugin version 0.2.0](https://github.com/ditrit/githubator-plugin/milestone/1) is released, we can integrated it in `leto-modelizer`.

### LINKED ISSUE #246: Multi-diagram view
## Description

We have to set up a new page that display all the diagrams of the projects.

We don't know how to access on this page, so we have to use url to be on this page.

url: `http://localhost:8080/#/[PROJECT_NAME]/diagrams`

We have to use the `DefaultNaviguationBar` for the header.

The purpose of this task is to have the page with header and all diagrams like the mockup:

image

### LINKED ISSUE #247: Multi-diagrams page: Zoom in/out
## Description

On the Multi-diagrams page, we want an action bar with two buttons: zoom in and zoom out.

The zoom feature is when we click on "zoom in" button we make a zoom on the center of all diagrams.
Same for zoom out, except it's a zoom, out ;).

image

### PR DESCRIPTION
**Pull request by k-y0u**
resolves #246 
resolves #247 
resolves #239 
resolves #240 
resolves #241

### CODE REVIEW

File: src/pages/DiagramsPage.vue [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import { useRoute } from 'vue-router';
+
+const route = useRoute();
+const projectName = computed(() => route.params.projectName);
+const diagrams = ref([]);
+const defaultFolder = ref(process.env.MODELS_DEFAULT_FOLDER !== ''
+  ? `${process.env.MODELS_DEFAULT_FOLDER}/`
+  : '');
+
+async function setDiagrams() {
```
Comment (Zorin95670) [inline src/pages/DiagramsPage.vue:38]: Documenation ?

File: cypress/e2e/DiagramsPage/display.diagrams.feature [hunk 2]
```diff
@@ -0,0 +1,58 @@
+Feature: Test diagrams page: display all diagrams
+
+  Scenario: Display a diagram
```
Comment (Zorin95670) [inline cypress/e2e/DiagramsPage/display.diagrams.feature:3]: Verify... Should ... Check ...

File: src/pages/DiagramsPage.vue [hunk 3]
```diff
@@ -8,6 +8,7 @@
       >
         <div
           :id="`${diagram.name}-diagram`"
+          :data-cy="`${diagram.name}-diagram`"
```
Comment (Zorin95670) [inline src/pages/DiagramsPage.vue:11]: image

`diagram_ID`

File: cypress/e2e/DiagramsPage/display.diagrams.feature (hunk 2 above)
Comment (k-y0u) [reply cypress/e2e/DiagramsPage/display.diagrams.feature]: Done

File: src/pages/DiagramsPage.vue (hunk 1 above)
Comment (k-y0u) [reply src/pages/DiagramsPage.vue:41]: Done

File: src/pages/DiagramsPage.vue (hunk 3 above)
Comment (k-y0u) [reply src/pages/DiagramsPage.vue]: Done

File: src/pages/DiagramsPage.vue [hunk 4]
```diff
@@ -0,0 +1,82 @@
+<template>
+  <q-page>
+    <div class="row q-gutter-md q-ma-md">
+      <q-card
+        v-for="diagram in diagrams"
+        :key="`${diagram.name}-diagram`"
+        class="diagram-container col-2 no-shadow q-pa-md bg-grey-3"
+      >
+        <div
+          :id="`diagram-${diagram.name}`"
```
Comment (justinehell) [inline src/pages/DiagramsPage.vue:10]: `${diagram.name}-diagram` line 6, use the same or change at line 6

Comment (justinehell) [reply src/pages/DiagramsPage.vue:10]: `${diagram.name}-diagram` used line 58 too

**Review by justinehell — CHANGES_REQUESTED** [review]

File: changelog.md [hunk 5]
```diff
@@ -17,6 +17,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   * Add tags list to filter project.
   * Add search bar for projects.
   * Add search bar for templates.
+* Multi diagrams page:
```
Comment (Zorin95670) [inline changelog.md:20]: Not on this line but, you have to delete the commit with the update dependencies

File: changelog.md [hunk 6]
```diff
@@ -17,6 +17,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+  * Display all diagrams of a project.
+  * Zoom center on all diagrams.
```
Comment (justinehell) [inline changelog.md:22]: `Zoom to center of all diagrams.` OR `Zoom centered on all diagrams.`

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/pages/DiagramsPage.vue [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+const defaultFolder = ref(process.env.MODELS_DEFAULT_FOLDER !== ''
+  ? `${process.env.MODELS_DEFAULT_FOLDER}/`
+  : '');
+
+/**
+ * Update diagrams list.
+ * @returns {Promise<void>} Promise with nothing on success otherwise an error.
+ */
+async function updateDiagrams() {
+  diagrams.value = await getAllModels(projectName.value);
```
Comment (justinehell) [inline src/pages/DiagramsPage.vue:86]: I don't get why you didn't need to mock `getAllModels` function ?

**Review by justinehell — CHANGES_REQUESTED** [review]

File: changelog.md (hunk 6 above)
Comment (k-y0u) [reply changelog.md:22]: Done

File: src/pages/DiagramsPage.vue [hunk 8]
```diff
@@ -14,6 +17,24 @@
...
+        <q-btn
+          icon="fa-solid fa-magnifying-glass-plus"
+          label="Zoom+"
+          stack
+          no-caps
+          @click="zoom(true)"
+        />
+        <q-btn
+          icon="fa-solid fa-magnifying-glass-minus"
+          label="Zoom-"
```
[the remark this answers is not in the document]
Comment (k-y0u) [reply src/pages/DiagramsPage.vue]: Done

File: src/pages/DiagramsPage.vue [hunk 9]
```diff
@@ -14,6 +17,24 @@
         />
       </q-card>
     </div>
+    <q-page-sticky :offset="[20, 20]">
+      <q-btn-group>
+        <q-btn
+          icon="fa-solid fa-magnifying-glass-plus"
+          label="Zoom+"
```
[the remark this answers is not in the document]
Comment (k-y0u) [reply src/pages/DiagramsPage.vue]: Done

File: cypress/e2e/DiagramsPage/display.diagrams.feature [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+    When I click on '[data-cy="component-definitions-item_terrator-plugin"]'
+    And  I click on '[data-cy="component-definition_aws"]'
+    Then I expect '[data-cy="modelizer-draw-view"] [data-cy="draw-container"]' exists
+    And  I expect '[id^="aws"]' exists
+
+    # Back to the models page
+    When I visit the 'localhost:8080/#/{{ projectName }}/diagrams'
+    Then I expect '[data-cy="diagram_{{ firstModelName }}"]' exists
+    Then I expect '[data-cy="diagram_{{ secondModelName }}"]' exists
```
[the remark this answers is not in the document]
Comment (k-y0u) [reply cypress/e2e/DiagramsPage/display.diagrams.feature]: Done

File: src/pages/DiagramsPage.vue [hunk 11]
```diff
@@ -24,13 +24,15 @@
           :label="$t('page.diagrams.actions.zoomPlus')"
           stack
           no-caps
+          data-cy="button_zoom-plus"
```
Comment (justinehell) [inline src/pages/DiagramsPage.vue:27]: zoom-plus-button

File: src/pages/DiagramsPage.vue [hunk 12]
```diff
@@ -24,13 +24,15 @@
...
           @click="zoom(true)"
         />
         <q-btn
           icon="fa-solid fa-magnifying-glass-minus"
           :label="$t('page.diagrams.actions.zoomMinus')"
           stack
           no-caps
+          data-cy="button_zoom-minus"
```
Comment (justinehell) [inline src/pages/DiagramsPage.vue:35]: zoom-minus-button

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/pages/DiagramsPage.vue (hunk 11 above)
Comment (k-y0u) [reply src/pages/DiagramsPage.vue]: done

File: README.md [hunk 13]
```diff
@@ -81,9 +81,9 @@ Plugin to manage github action files.
     </tr>
     <tr>
       <td>next version</td>
-      <td>= 0.15.2</td>
```
Comment (Zorin95670) [inline README.md:84]: No no no no no
Keep this line and add a new one with the new values

**Review by Zorin95670 — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

File: src/pages/DiagramsPage.vue [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+  );
+}
+
+onMounted(async () => {
+  await updateDiagrams();
+  await drawDiagrams();
+});
+</script>
+
+<style lang="scss" scoped>
```
Comment (Zorin95670) [inline src/pages/DiagramsPage.vue:115]:
```html
<style lang="scss" scoped>
.diagram-container {
  display: inline-block;
  border: 2px dashed $primary;
}
</style>
<style lang="scss">
.diagram-container div svg {
    overflow: visible !important;
    display: unset;
    height: 100%;
    width: 100%;
  }
</style>
```

**Pull request merged**